core/store/orm/locking_strategies.go

```diff
@@ -119,30 +133,76 @@ func (s *PostgresLockingStrategy) Lock(timeout time.Duration) error {
 	}
 	s.db = db
 
+	// `database/sql`.DB does opaque connection pooling, but PG advisory locks are per-connection
+	conn, err := db.Conn(ctx)
+	if err != nil {
+		return err
+	}
+
+	s.conn = conn
+
+	return nil
+}
+
+func (s *PostgresLockingStrategy) Close(timeout time.Duration) error {
+	s.m.Lock()
+	defer s.m.Unlock()
+
 	ctx := context.Background()
 	if timeout != 0 {
 		var cancel context.CancelFunc
 		ctx, cancel = context.WithTimeout(ctx, timeout)
 		defer cancel()
 	}
-	_, err = s.db.ExecContext(ctx, "SELECT pg_advisory_lock($1)", postgresAdvisoryLockID)
+
+	_, err := s.conn.ExecContext(ctx, "SELECT pg_advisory_unlock_all()")
```

Comment:
What if we add other advisory locks in the future? Will this unlock all of them?

Reply:
Whoops, good catch.  Was using that to debug the race condition.